### PR TITLE
Terminal editors

### LINKED ISSUE #126244: Terminal Editors: Create TerminalEditor class
Sub-task of https://github.com/microsoft/vscode/issues/125514

- Just connect to a terminal view process until the `TerminalGroupService` is ready (https://github.com/microsoft/vscode/issues/126243)
- Can this file be lazy loaded as it's an optional feature?

### PR DESCRIPTION
**Pull request by meganrogge**
fixes #126244

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/media/terminal.css [hunk 1]
```diff
@@ -17,41 +17,49 @@
...
 	height: 100%;
 }
 
+.monaco-workbench .editor-instance .terminal-wrapper,
 .monaco-workbench .pane-body.integrated-terminal .terminal-wrapper {
 	display: none;
 	margin: 0 10px;
 	bottom: 2px;
 }
+
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/media/terminal.css:32]: add this everywhere below

File: src/vs/workbench/contrib/terminal/browser/terminal.ts [hunk 2]
```diff
@@ -90,6 +91,29 @@ export const enum TerminalConnectionState {
...
+	Editor = 'editor'
+}
+
+export interface ICreateTerminalOptions {
+	/**
+	 * The shell launch config or profile to launch with, when not specified the default terminal
+	 * profile will be used.
+	 */
+	config?: IShellLaunchConfig | ITerminalProfile;
+	// TODO: Ensure this is supported for profiles
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminal.ts:105]: remove

File: src/vs/workbench/contrib/terminal/browser/terminalActions.ts [hunk 3]
```diff
@@ -191,6 +191,39 @@ export function registerTerminalActions() {
...
+				id: TerminalCommandId.CreateTerminalEditor,
+				title: { value: localize('workbench.action.terminal.createTerminalEditor', "Create Terminal Editor"), original: 'Create Terminal Editor' },
+				f1: true,
+				category,
+				precondition: KEYBINDING_CONTEXT_TERMINAL_PROCESS_SUPPORTED
+			});
+		}
+		async run(accessor: ServicesAccessor) {
+			const terminalService = accessor.get(ITerminalService);
+			// TODO: Await openEditor
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalActions.ts:206]: add @Tyriar

File: src/vs/workbench/contrib/terminal/browser/terminalEditor.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	override setVisible(visible: boolean, group?: IEditorGroup): void {
+		super.setVisible(visible, group);
+
+		if (!this._editorInput?.terminalInstance) {
+			return;
+		}
+
+		this._editorInput.terminalInstance.setVisible(visible);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalEditor.ts:57]: ? after editoInput and instance and delete the above

File: src/vs/workbench/contrib/terminal/browser/terminalEditor.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	constructor(
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalEditor.ts:60]: move to below props

File: src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		super();
+		this._terminalInstance = terminalInstance;
+		this._terminalInstance.onTitleChanged(() => this._onDidChangeLabel.fire());
+	}
+
+	override getName() {
+		return this.terminalInstance.title;
+	}
+
+	override dispose() {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalEditorInput.ts:40]: this.register(toDisposable(this.terminalInstance.dispose) in constructor and delete this

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 7]
```diff
@@ -639,6 +639,12 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		this._container.appendChild(this._wrapperElement);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:639]: delete reattach

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 8]
```diff
@@ -277,7 +278,10 @@ export class TerminalService implements ITerminalService {
 					terminalLayouts.forEach((terminalLayout) => {
 						if (!terminalInstance) {
 							// create group and terminal
-							terminalInstance = this.createTerminal({ attachPersistentProcess: terminalLayout.terminal! });
+							const config = { attachPersistentProcess: terminalLayout.terminal! } as IShellLaunchConfig;
+							terminalInstance = this.createTerminal(
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:282]: inline

File: src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts [hunk 9]
```diff
@@ -107,7 +107,7 @@ export class TerminalTabList extends WorkbenchList<ITerminalInstance> {
 		this._terminalService.onDidChangeConnectionState(() => this.refresh());
 		this._themeService.onDidColorThemeChange(() => this.refresh());
 		this._terminalService.onActiveInstanceChanged(e => {
-			if (e) {
+			if (e && e.target !== TerminalTarget.Editor) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalTabsList.ts:110]: create issue to get rid of this when group service is added

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 8 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:282]: didn't work. going w it as is

**Comment by Tyriar** [comment]
Notebook and smoke tests failing, merging this in

**Pull request merged**